addons/web/static/src/core/barcode/barcode_video_scanner.js

```diff
@@ -112,10 +121,13 @@ export class BarcodeVideoScanner extends Component {
      */
     async isVideoReady() {
         // FIXME: even if it shouldn't happened, a timeout could be useful here.
-        while (!isVideoElementReady(this.videoPreviewRef.el)) {
+        while (this.videoPreviewRef.el && !isVideoElementReady(this.videoPreviewRef.el)) {
             await delay(10);
         }
         this.state.isReady = true;
```

Comment:
that's not really true. You might come here because the previewRef doesn't exist anymore, and in this case it's not ready